subsys/bluetooth/audio/shell/hap_harc.c

```diff
@@ -0,0 +1,596 @@
+		return -ENOEXEC;
+	}
+
+	harc[0] = harc_insts[bt_conn_index(default_conn)];
+	if (harc[0] == NULL) {
+		shell_error(sh, "%s: %p not bound",
+			    sh->ctx->active_cmd.syntax, (void *)default_conn);
+		return -ENOEXEC;
+	}
+
+	preset_write_params.complete = NULL;
+	preset_write_params.status = NULL;
+	preset_write_params.index = shell_strtoul(argv[1], 16, &err);
+	if (err < 0) {
+		shell_error(sh, "Invalid index parameter (err %d)", err);
+		return err;
+	}
+
+	preset_write_params.name = argv[2];
+
+	err = bt_hap_harc_info_get(harc[0], &info);
+	if (err != 0) {
+		shell_error(sh, "bt_hap_harc_info_get (err %d)", err);
+		return -ENOEXEC;
+	}
+
+	if (info.type == BT_HAS_HEARING_AID_TYPE_BINAURAL && info.binaural.pair != NULL) {
+		harc[1] = info.binaural.pair;
+		count++;
+	}
+
+	err = bt_hap_harc_preset_write(harc, count, &preset_write_params);
+	if (err != 0) {
+		shell_error(sh, "bt_hap_harc_preset_write (err %d)", err);
+		return -ENOEXEC;
+	}
+
+	return err;
+}
+
+static int cmd_harc(const struct shell *sh, size_t argc, char **argv)
+{
+	if (argc > 1) {
+		shell_error(sh, "%s unknown parameter: %s", argv[0], argv[1]);
+	} else {
+		shell_error(sh, "%s missing subcomand", argv[0]);
+	}
+
+	return -ENOEXEC;
+}
+
+static void harc_preset_proc_complete_cb(int err, void *params)
+{
+	if (params == &preset_read_params) {
+		shell_print(ctx_shell, "Preset read complete (err %d)", err);
+	} else if (params == &preset_write_params) {
+		shell_print(ctx_shell, "Preset write complete (err %d)", err);
+	} else if (params == &preset_set_params) {
+		shell_print(ctx_shell, "Preset set complete (err %d)", err);
+	} else {
+		shell_error(ctx_shell, "Unexpected params %p", params);
+	}
+}
+
+static void harc_preset_proc_status_cb(struct bt_hap_harc *harc, int err, void *params)
+{
+	if (params == &preset_read_params) {
+		shell_print(ctx_shell, "HARC %p Preset read status (err %d)", harc, err);
+	} else if (params == &preset_write_params) {
+		shell_print(ctx_shell, "HARC %p Preset write status (err %d)", harc, err);
+	} else if (params == &preset_set_params) {
+		shell_print(ctx_shell, "HARC %p Preset set status (err %d)", harc, err);
+	} else {
+		shell_error(ctx_shell, "HARC %p Unexpected params %p", harc, params);
+	}
+}
+
+#define HELP_NONE "[none]"
+
+SHELL_STATIC_SUBCMD_SET_CREATE(harc_cmds,
+	SHELL_CMD_ARG(init, NULL, HELP_NONE, cmd_harc_init, 1, 0),
+	SHELL_CMD_ARG(bind, NULL, HELP_NONE, cmd_harc_bind, 1, 0),
+	SHELL_CMD_ARG(unbind, NULL, HELP_NONE, cmd_harc_unbind, 1, 0),
+	SHELL_CMD_ARG(preset-read, NULL, "<start_index> <max_count>", cmd_harc_preset_read, 3, 0),
+	SHELL_CMD_ARG(preset-set, NULL, "<index>", cmd_harc_preset_set, 2, 0),
+	SHELL_CMD_ARG(preset-next, NULL, HELP_NONE, cmd_harc_preset_set_next, 1, 0),
+	SHELL_CMD_ARG(preset-prev, NULL, HELP_NONE, cmd_harc_preset_set_prev, 1, 0),
+	SHELL_CMD_ARG(preset-write, NULL, "<index> <name>", cmd_harc_preset_write, 3, 0),
+	SHELL_SUBCMD_SET_END
+);
```

Comment:
Can you add a `hap.rst` next to the other shell documentation files with a brief explanation on how to use (at least some of) these commands?